android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetOngoingConferenceService.java

```diff
@@ -95,26 +100,48 @@ private static void doLaunch(Context context, HashMap<String, Object> extraData)
 
 
     public static void launch(Context context, HashMap<String, Object> extraData) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            PermissionListener listener = new PermissionListener() {
-                @Override
-                public boolean onRequestPermissionsResult(int i, String[] strings, int[] results) {
-                    if (results.length > 0 && results[0] == PackageManager.PERMISSION_GRANTED) {
-                        doLaunch(context, extraData);
-                    }
-
-                    return true;
+        List<String> permissionsList = new ArrayList<String>();
+
+
+        PermissionListener listener = new PermissionListener() {
+            @Override
+            public boolean onRequestPermissionsResult(int i, String[] strings, int[] results) {
+                if (results.length > 0 && results[0] == PackageManager.PERMISSION_GRANTED) {
+                    JitsiMeetLogger.w(TAG + " POST_NOTIFICATIONS permission granted");
                 }
-            };
+
+                return true;
+            }
+        };
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            permissionsList.add(POST_NOTIFICATIONS);
 
             JitsiMeetActivityDelegate.requestPermissions(
                 (Activity) context,
                 new String[]{POST_NOTIFICATIONS},
                 POST_NOTIFICATIONS_PERMISSION_REQUEST_CODE,
                 listener
             );
-        } else {
-            doLaunch(context, extraData);
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            permissionsList.add(RECORD_AUDIO);
+        }
+
+        String[] permissionsArray = new String[ permissionsList.size() ];
+        permissionsList.toArray( permissionsArray );
+
+        for (String permission : permissionsList) {
+            boolean checkPermission
+                = ContextCompat.checkSelfPermission(context, permission)
+                == PackageManager.PERMISSION_GRANTED;
+
+            if (checkPermission) {
```

Comment:
This is wrong, it will pass if one of the permissions is granted but not the other one. Don;t check, just request them direcrtly.